src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -139,3 +140,17 @@ def _lock_cb(cls, mode, n, file, line):
                     mode, n, file, line
                 )
             )
+
+
+# https://github.com/pyca/cryptography/issues/2299 contains extensive
+# discussion about this workaround. In essence, we are acquiring a global lock
+# (including across all subinterpreters if we're in a mod_wsgi environment)
+# initializing some C level locks, then initializing the binding and adding the
+# OS random engine. This should prevent engine addition race conditions,
+# although it's still unclear what happens when a subinterpreter which
+# has registered the locks is destroyed. Do the callback functions still work
+# or are all the objects associated with that subinterpreter destroyed?
+imp.acquire_lock()
+Binding.init_static_locks()
+Binding._ensure_ffi_initialized()
+imp.release_lock()
```

Comment:
This should definitely be inside of a `finally` block. I still need to think of any of this makes sense though :-)